File: project_euler/Problem064/Problem064.js
function Problem064() {
    let target = 10000
    let result = 0
    for (let i = 1; i <= target; ++i) {
        if (0 !== (getPeriod(i) & 1)) {
            ++result
        }
    }
    console.log(result)
}

function getPeriod(value) {
    let auxiliary = Math.sqrt(value)
    if (Math.abs(Math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001) {
        return 0
    }

    let initial = Math.floor(auxiliary)
    let pivot = initial << 1
    let current = initial
    let offset = 0
    let factor = 1
    let count = 0
    while (current !== pivot) {
        offset = current * factor - offset
        factor = Math.floor((value - offset * offset) / factor)
        current = Math.floor((initial + offset) / factor)
        ++count
    }
    return count
}

module.exports = {
    Problem064 : Problem064
};

File: project_euler/Problem064/Problem064.go
package project_euler

import (
    "fmt"
    "math"
)

func Problem064() {
    target := 10000
    result := 0
    for i := 1; i <= target; i++ {
        if 0 != (getPeriod(i) & 1) {
            result++
        }
    }
    fmt.Println(result)
}

func getPeriod(value int) int {
    auxiliary := math.Sqrt(float64(value))
    if math.Abs(math.Floor(auxiliary + 0.5) - auxiliary) < 0.00000001 {
        return 0
    }

    initial := int(auxiliary)
    pivot := initial << 1
    current := initial
    offset := 0
    factor := 1
    count := 0
    for current != pivot {
        offset = current * factor - offset
        factor = (value - offset * offset) / factor
        current = (initial + offset) / factor
        count++
    }
    return count
}
